src/components/kubectl/port-forward.ts

```diff
@@ -264,11 +315,38 @@ function buildPortPair(portPair: string): PortMapping {
     const targetPort = splitString[1];
 
     return {
-        localPort: Number(localPort),
-        targetPort: Number(targetPort)
+        localPort: buildNullablePort(validPorts, localPort),
+        targetPort: buildPort(validPorts, targetPort)
     };
 }
 
+/**
+ * Builds a single numberic port for a PortMapping object from the captured user input allowing empty or zero value
+ * @param validPorts List of valid named ports
+ * @param portString The port provided by the user
+ * @returns numberic port number
+ */
+function buildNullablePort(validPorts: ExtractedPort[], port: string): number | undefined {
+    if (['', '0'].includes(port)) {
+        return undefined;
+    }
+    return buildPort(validPorts, port);
+}
+
+/**
+ * Builds a single numberic port for a PortMapping object from the captured user input
+ * @param validPorts List of valid named ports
+ * @param portString The port provided by the user
+ * @returns numberic port number
+ */
+function buildPort(validPorts: ExtractedPort[], port: string): number {
+    const found = validPorts.filter(({name}) => name === port);
```

Comment:
Consider using `find`:

```suggestion
    const validPort = validPorts.find(({name}) => name === port);
```

(and then `if (validPort)`)